invoice.py

```diff
@@ -1576,9 +1580,19 @@ def generate_edi(self):
             with open(template_path) as file_:
                 template = Template(file_.read())
             edi_file = template.render({'invoice': self})
-            if not os.path.exists(result_path):
-                with open(result_path, 'w', encoding='latin-1') as f:
-                    f.write(edi_file)
+
+            for count in range(retry, -1, -1):
+                if not os.path.exists(result_path):
+                    try:
+                        with open(result_path, 'w', encoding='latin-1') as f:
+                            f.write(edi_file)
+                        break
+                    except OSError:
+                        if not count:
+                            raise
+                        else:
+                            time.sleep(0.02 * (retry - count))
+                    break
```

Comment:
This `break` must be removed as it will avoid the iteration

Reply:
See changeset 0cd7d244406de02184da0198f857dabcc15a8891